File: backend/app/schemas/state_context.py
from typing import Literal

from backend.app.schemas.upload_preview import ApiModel


StateContextClass = Literal[
    "operator_package",
    "development_default",
    "qa_temporary",
    "configured",
    "unknown",
    "inaccessible",
]
StateStorageStatus = Literal["present", "missing", "inaccessible", "unknown"]
StateContextSource = Literal["default", "env", "init", "unknown"]


class StateContextDto(ApiModel):
    context_class: StateContextClass
    label: str
    storage_status: StateStorageStatus
    source: StateContextSource


def unknown_state_context() -> StateContextDto:
    return StateContextDto(
        context_class="unknown",
        label="Unknown state",
        storage_status="unknown",
        source="unknown",
    )

File: frontend/src/api/stateContext.ts
export type StateContextClass =
  | "operator_package"
  | "development_default"
  | "qa_temporary"
  | "configured"
  | "unknown"
  | "inaccessible"
  | string;

export type StateStorageStatus = "present" | "missing" | "inaccessible" | "unknown" | string;

export interface StateContext {
  contextClass: StateContextClass;
  label: string;
  storageStatus: StateStorageStatus;
  source: "default" | "env" | "init" | "unknown" | string;
}

export const unknownStateContext: StateContext = {
  contextClass: "unknown",
  label: "Unknown state",
  storageStatus: "unknown",
  source: "unknown",
};
